billy-portugal/src/main/java/com/premiumminds/billy/portugal/persistence/dao/jpa/AbstractDAOPTGenericInvoiceImpl.java

```diff
@@ -44,10 +44,13 @@ public TInterface findByNumber(UID uidBusiness, String number) {
         QJPAPTGenericInvoiceEntity invoice = QJPAPTGenericInvoiceEntity.jPAPTGenericInvoiceEntity;
 
         return (TInterface) this.checkEntity(
-                this.createQuery()
-                        .from(invoice).where(this.toDSL(invoice.business, QJPAPTBusinessEntity.class).uid
-                                .eq(uidBusiness.toString()).and(invoice.number.eq(number)))
-                        .singleResult(invoice),
-                PTGenericInvoiceEntity.class);
+            this.createQuery()
+                .from(invoice)
+                .where(this.toDSL(invoice.business, QJPAPTBusinessEntity.class).uid
+                           .eq(uidBusiness.toString())
+                           .and(invoice.number.eq(number)))
+                .select(invoice)
+                .fetchOne(),
```

Comment:
.fetchFirst() ?

Reply:
As opposed to what we discussed earlier on the project with the library usage, here it was decided to also fix the behavior. 
Just the migration would be `fetchFirst()`, as you sugested.